contrib/babelfishpg_common/sql/upgrades/spatial_types--3.3.0--3.4.0.sql

```diff
@@ -590,9 +596,31 @@ CREATE OR REPLACE FUNCTION sys.bytea_helper(sys.GEOMETRY)
 	AS '$libdir/postgis-3','LWGEOM_to_bytea'
 	LANGUAGE 'c' IMMUTABLE STRICT PARALLEL SAFE;
 
+CREATE OR REPLACE FUNCTION sys.charTogeomhelper(sys.bpchar)
+	RETURNS sys.GEOMETRY
+	AS $$
+	DECLARE
+		Geomtype text;
+		geom sys.GEOMETRY; 
+	BEGIN
+		-- Call the underlying function after preprocessing
+		geom = (SELECT sys.stgeomfromtext_helper($1, 0));
+		Geomtype = (SELECT sys.ST_GeometryType(geom));
+		IF Geomtype = 'ST_Point' THEN
+			IF (SELECT sys.ST_Zmflag(geom)) = 1 OR (SELECT sys.ST_Zmflag(geom)) = 2 OR (SELECT sys.ST_Zmflag(geom)) = 3 THEN
```

Comment:
1. Let's store the value of `ST_Zmflag` first and use it everywhere instead of calling the function multiple times.
2. Can you explain the meaning of values 1,2,3? Maybe add a comment.

Reply:
Done